### ISSUE TITLE
Crayon: Swap to SConstruct

### ISSUE THREAD

**Issue by Protofall** [Protofall/Crayon-Game-Framework#105]
The Makefile format has proven to have too many faults, some that can't be worked around. Such issues like Issue #50 can't be fully fixed due to a bug in how "include" works. Then there's code duplication for cd and sd builds (And later pc with BBAs) due to the fact I can't provide rules a variable for a compiler flag. Finally there's the fact its hard to debug Makefiles and it takes me a very long time to get a basic setup going, infact JamoHTP made the current iteration of Crayon's Make system.

Because of this I plan to learn and swap over to **SConstruct**, a Python-based more advanced building tool. It should be a lot more flexible and easy to work with. As part of learning how it works I will convert `KallisitiOSTests/Kmg` to be built with SConstruct. I'll keep its makefile version in there aswell for those who don't want to bother with installing SConstruct. Once that is done I'll work on adapting Crayon's Make system.

**Comment by Protofall** [comment]
#236 is kind of a duplicate of this, but I'll leave it open since its a bit more specific in regards to the Crayon-Savefile repo stuff

**Comment by Protofall** [comment]
As of now, in branch `#105-#236-#231-Scons-Update`, I have a system that fully builds the libraries with SContruct and also the scons file in `Camera-Textured` demo is able to build the library as well. Here's what's left:
- Create builders for `elfToBin`, `scrambled bin`, `iso`, `cdi` (So we can re-use them for each different build)
- Call the preprocessing script from our python code. Figure out how I should remove stuff in order for it to build correctly
   - For now, just skip this. Use the "Lines" demo to make sure the programs are built right since that doesn't use assest
- Figure out the "build" folder structure for a project. I thought I made notes somewhere, but I don't see them here
- Copy the scons files over to all demos and delete all old scons/makefile stuff (Including `Crayon/rules`)
- Rename `Crayon` dir to `libCrayon` similar to have Crayon-Savefile has done.

**Comment by Protofall** [comment]
I think the idea with the build structure is it will be like this.
- `ls build/`: All the different systems (`dreamcast`, `pc`). Each of those will contain the different builds for that platform. For example dreamcast will have `dreamcast-cd`, `dreamcast-sd`, `dreamcast-pc` and later when we add PC support there will also be `windows`, `linux` etc. Each of the folders in `build/` will contain their own `cdfs/` folder. This is because all dreamcast boot modes pretty much share the same assets.
   - NOTE: The cdfs/ folder will contain a `1ST_READ.BIN` from the cd build, however the fact that this is here won't prevent the sd and pc versions from working since you'll need to add the binary later
- Each of those boot mode folders will contain all the intermediates used to make it. (`.o`, `.elf`, `.bin (scrambled)`, `.iso`, `cdfs/` folder)
- If we go back to the project root and do `ls programs/` we will see all the different builds in one folder. dreamcast-cd will have the cdi whereas sd and pc will have their corresponding unscrambled binaries as well as a aliases/shortcut to the `cdfs/` folder
